SYCL/GroupLocalMemory/group_local_memory.cpp

```diff
@@ -0,0 +1,95 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
```

Comment:
New directory is added. Could you please set correct code owner (in .github/CODEOWNERS)

Reply:
Added myself as the code owner. @Pennycook Do you mind being set as the second code owner for the group local memory tests? (I this can be done as a separate PR, so that it does not block merging this one).

Reply:
> @Pennycook Do you mind being set as the second code owner for the group local memory tests?

Fine by me.